ocpp_codec/serializer.py

```diff
@@ -400,7 +400,10 @@ def serialize_fields(message):
             field_value = _run_validator(field, field_value, parsing=False)
             # Serialize the list's elements
             field = _unpack_field(field)
-            serialize_func = serialize_fields if issubclass(field.type, types.ComplexType) else serialize_field
+            if issubclass(field.type, types.ComplexType):
+                serialize_func = serialize_fields
+            else:
+                serialize_func = functools.partial(serialize_field, field)
```

Comment:
I don't really understand the problem this is solving. You seem to had a restriction on `optionalComplexListValue` number of elements, but then in your test you test only the validation of the elements it contains?

Reply:
I've split the commit in two and reworded both commit messages.
Could you tell me if it's clearer now? :)